components/UserTestimonies/Tabs.tsx

```diff
@@ -0,0 +1,127 @@
+import React, {
+  MouseEventHandler,
+  ReactElement,
+  useEffect,
+  useRef,
+  useState
+} from "react"
+import styled from "styled-components"
+
+type onClickEventFunction = (e: Event, value: number) => void
+
+export const TabSlider = (props: { width: number; position: number }) => {
+  const { width, position } = props
+  return <TabSliderStyle width={width ?? 200} position={position} />
+}
+
+export const TabSliderContainer = (props: { children?: JSX.Element }) => {
+  const { children } = props
+  return <TabSliderContainerStyle>{props.children}</TabSliderContainerStyle>
+}
+
+export const Tab = (props: {
+  label: string
+  value: number
+  active: boolean
+  onClick?: MouseEventHandler
+}) => {
+  const { label, onClick, active } = props
+  return (
+    <TabStyle onClick={onClick} active={active}>
+      <h3> {label}</h3>
+    </TabStyle>
+  )
+}
+
+export const Tabs = (props: {
+  childTabs: ReactElement[]
+  onChange: onClickEventFunction
+  selectedTab: number
+}) => {
+  const containerRef = useRef(null)
+  const tabRefs = useRef<Array<HTMLDivElement | null>>([])
+  const { childTabs, onChange, selectedTab } = props
+  const [sliderWidth, setSliderWidth] = useState(0)
+  const [sliderPos, setSliderPos] = useState(0)
+  const sliderPositionOffset = 16
+
+  useEffect(() => {
+    if (
+      tabRefs.current[selectedTab - 1] !== null &&
```

Comment:
nit: define a local variable in this effect to reduce duplication and check for non-nullness:

```
const selected = tabRefs.current[selectedTab - 1]
if (selected?.getBoundingClientRect().width) {
  setSliderWidth(selected.getBoundingClientRect().width)
  setSliderPos(selected.getBoundingClientRect().x)
}
```